lib/net/ldap/connection.rb

```diff
@@ -52,7 +52,8 @@ def open_connection(server)
     hosts.each do |host, port|
       begin
         prepare_socket(server.merge(socket: @socket_class.new(host, port, socket_opts)), timeout)
-        if encryption[:tls_options][:verify_mode] != OpenSSL::SSL::VERIFY_NONE
+        if encryption && encryption[:tls_options] &&
```

Comment:
I re-did this so it's a better copy of #259, but the nil check remains. I think you're right, it should return at least an empty hash. Next PR?